From c9d0094a50f25d09eb74c02dcd57494d62287838 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH 151/296] Google issue: 64372366 the mAvailableWidth is not
 right when rotate screen.

Referring to 64372366 for detailed information. After rotating the
screen to vertical, the mAvailiableWidth still is the value of
landscape. But the actual value is roght. So we use metrics.widthPixels.
---
 core/java/com/android/internal/policy/DecorView.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/java/com/android/internal/policy/DecorView.java b/core/java/com/android/internal/policy/DecorView.java
index 4e06577e358..e069c749584 100644
--- a/core/java/com/android/internal/policy/DecorView.java
+++ b/core/java/com/android/internal/policy/DecorView.java
@@ -735,7 +735,7 @@ public class DecorView extends FrameLayout implements RootViewSurfaceTaker, Wind
                 if (tv.type == TypedValue.TYPE_DIMENSION) {
                     min = (int)tv.getDimension(metrics);
                 } else if (tv.type == TypedValue.TYPE_FRACTION) {
-                    min = (int)tv.getFraction(mAvailableWidth, mAvailableWidth);
+                    min = (int)tv.getFraction(metrics.widthPixels, metrics.widthPixels);
                 } else {
                     min = 0;
                 }
-- 
2.11.0

